airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/JsonSchemaType.java

```diff
@@ -17,7 +17,7 @@ public enum JsonSchemaType {
 
   STRING("string", true, null, Schema.Type.STRING),
   NUMBER_INT("number", true, "integer", Schema.Type.INT),
```

Comment:
I think this (and the `INTEGER` entry below) should be switched to LONG, but had a _ton_ of trouble actually making that change. Given that current behavior is both wrong and unused, I feel fine with pushing this off until we finish the types revamp project. Given the uncertainty around what the final state looks like, will hold off on writing an issue for now.

---

for the curious: The problem is that we have some tests which test that we can handle unions of timestamp and integer. This resolves to an Avro union with types `["long", {"type": "long", "logicalType": "timestamp-micros"}]`. Unfortunately, Avro doesn't support this union, and complains that `long` is a duplicate entry.

But we also can't just drop one of those entries, because Avro expects a numeric input for `long`, but string input for the timestamp-micros. In other words:
* `{"created_at": 1634982000}` can be parsed as `"long"`, but not as `{"type": "long", "logicalType": "timestamp-micros"}`
* `{"created_at": "1634982000"}` (note the quotes around `1634982000`) can be parsed as `{"type": "long", "logicalType": "timestamp-micros"}`, but not as `"long"`.

Reply:
@edgao it was having problem with bigint. 

postgres type int8 (bigint)
airbyte schema  ended up- {"type": "number", "airbyte_type": "integer"}

in parquet ended up integer and long values trimmed down and data is unusable. i had to patch to Long to get it work.